crates/server/src/admin/handlers/context/get_context_identities.rs

```diff
@@ -15,39 +17,55 @@ pub async fn handler(
     Extension(state): Extension<Arc<AdminState>>,
     req: Request,
 ) -> impl IntoResponse {
-    let context = state
-        .ctx_client
-        .get_context(&context_id)
-        .map_err(|err| parse_api_error(err).into_response());
-
-    let context = match context {
-        Ok(context) => context,
-        Err(err) => return err.into_response(),
-    };
+    let owned = req.uri().path().ends_with("identities-owned");
 
-    let context = match context {
-        Some(context) => context,
-        None => {
+    // Get the context info
+    let context = match state.ctx_client.get_context(&context_id) {
+        Ok(Some(context)) => context,
+        Ok(None) => {
             return ApiError {
                 status_code: StatusCode::NOT_FOUND,
                 message: "Context not found".into(),
             }
             .into_response()
         }
+        Err(err) => return parse_api_error(err).into_response(),
     };
 
-    let owned = req.uri().path().ends_with("identities-owned");
+    // Clone what we need
+    let id = context.id;
+    let ctx_client = state.ctx_client.clone();
+
+    // Process the stream in a blocking task to handle non-Send types
```

Comment:
Looks okay to me, only small thing I noticed a lot of code comment and not sure if we need all of this commented cc @miraclx